RFC/src/RFC-0241_AtomicSwapXMR.md

```diff
@@ -0,0 +1,789 @@
+
+
+### Commitment phase
+
+This phase allows Alice and Bob to commit to using their keys and requires more than one round to complete.
+Some of the information that needs to be committed to depends on previous knowledge. 
+
+Alice needs to provide Bob with the following:
+
+* Script public key: \\( K_{Sa}\\)
+* Monero public key:  \\( Xm_a\\)
+
+$$
+\begin{aligned}
+Xm_a &=  x_a \cdot M \\\\
+\end{aligned}
+\tag{3}
+$$
+
+Bob needs to provide Alice with the following:
+
+* Script public key: \\( K_{Sb}\\)
+* Monero public key:  \\( Xm_b\\)
+
+$$
+\begin{aligned}
+Xm_b &=  x_b \cdot M \\\\
+\end{aligned}
+\tag{4}
+$$
+
+### XTR payment
+
+Alice will construct the Tari UTXO with the correct [script](#tariscript) and publish the containing transaction to the 
+blockchain, knowing that she can reclaim her Tari if Bob vanishes or tries to break the agreement. This is done with 
+standard Mimblewimble rules and signatures.
+
+### XMR payment
+
+When Bob sees that the Tari UTXO that Alice created is mined on the Tari blockchain with the correct script, Bob can go ahead 
+and publish the Monero transaction containing the Monero UTXO with the aggregate key \\(Xm = Xm_a + Xm_b \\).
+
+### Claim XTR 
+
+When Alice sees that the Monero UTXO that Bob created is mined on the Monero blockchain containing the correct aggregate 
+key \\(Xm\\), she can provide Bob with the requird `pre_image` to spend the Tari UTXO. She does not have the 
+missing key \\(x_b \\) to claim the Monero yet, but it will be revealed when Bob claims the Tari. 
+
+Bob can now supply the `pre_image` and his Monero private key as transaction input to unlock the script.
+
+### Claim XMR
+
+Alice can now see that Bob spent the Tari UTXO, and by examining the `input_data` required to satisfy the script, she 
+can learn Bob's secret Monero key. Although this private key is now public knowledge, her part of the Monero spend key 
+is still private, and thus only she knows the complete Monero spend key. She can use this knowledge to claim the Monero 
+UTXO.
+
+### The refund
+
+If something goes wrong and Bob never publishes the Monero, or he disappears, Alice needs to wait for the lock height
+`height_1` to pass. This will allow her to reclaim her Tari, but in doing so, she needs to publish her Monero secret key 
+as input to the script to unlock the Tari. When Bob comes back online, he can use this public knowledge to reclaim his 
+Monero as only he now knows both parts of the Monero UTXO spend key.
+
+
+### The lapse transaction
+
+If something goes wrong and Alice never gives Bob the requird `pre_image`, Bob needs to wait for the lock height
+`height_2` to pass. This will allow him to claim the Tari he wanted all along, but in doing so, he needs to publish
+his Monero secret key as input to the script to unlock the Tari. When Alice comes back online, she can use this public 
+knowledge to claim the Monero she wanted all along as only she now knows both parts of the Monero UTXO spend key.
+
```

Comment:
I think we need to add a section here to the like of **Script security** that discusses the security aspects of revealing half the Monero private key and why that is not a security risk, with some informed/original references to back it up. 

Then we can forward reference this section in all parts where Alice or Bob must reveal half their Monero private key to the public.

Reply:
I don't quite feel that we can go deep enough for it to be worth it. To properly explain why this is secure we need to explain ECC. And that's not something that we should do here. 

If I just add a section that mentions this, will it be enough for you:
" Because of the way Elliptic Curve Cryptography works we know that we can add two secret keys together, and share the public version of both. And at the same time, we know that neither party can calculate the total secret key with just one part. "

Reply:
That could work, but I think we would still need some informed/original references to back it up.

Reply:
Added, do you have a good reference that explains this?

Reply:
Let me try to hunt something down.

Reply:
I still think this issue will be in people's minds thus it warrants a separate section, and not just added into a paragraph as a type of afterthought.

Maybe something like this:

_**Script security**_
_The risk of any of our two parties sharing half the Monero private key with their counterparty is similar to the 
security of the Elliptic-Curve Cryptography scheme 
[as summarized here](https://cryptobook.nakov.com/asymmetric-key-ciphers/elliptic-curve-cryptography-ecc#private-key-public-key-and-the-generator-point-in-ecc). 
Considering (2) and (3) we have:_

```
  (x_a + x_b) . M = x_a . M + x_b . M     (5)
                  = Xm_a + Xm_b
    with
  unknown: x_a
  known:   x_b, (x_a + x_b) . M, Xm_a, Xm_b
```
_We can therefore be led to believe that (6) is feasible:_
```
  x_a = (Xm_a + Xm_b - x_b . M) / M       (6)
      = Xm_a / M
```
_However, this is exactly equivalent to solving the 
[Elliptic-Curve Discrete Logarithm Problem (ECDLP)](https://cryptobook.nakov.com/asymmetric-key-ciphers/elliptic-curve-cryptography-ecc#elliptic-curve-discrete-logarithm-problem-ecdlp), 
which have been shown to have no efficient solution for carefully chosen finite fields and elliptic curves._